checkov/common/runners/base_runner.py

```diff
@@ -125,16 +126,19 @@ def load_external_checks(self, external_checks_dir: List[str]) -> None:
     def get_graph_checks_report(self, root_folder: str, runner_filter: RunnerFilter) -> Report:
         return Report(check_type="not_defined")
 
-    def run_graph_checks_results(self, runner_filter: RunnerFilter, report_type: str) -> dict[BaseGraphCheck, list[_CheckResult]]:
+    def run_graph_checks_results(self, runner_filter: RunnerFilter, report_type: str, graph: Graph | None = None
+                                 ) -> dict[BaseGraphCheck, list[_CheckResult]]:
         checks_results: "dict[BaseGraphCheck, list[_CheckResult]]" = {}
-        if not self.graph_manager or not self.graph_registry:
+        if graph is None and (not self.graph_manager or not self.graph_registry):
             # should not happen
             logging.warning("Graph components were not initialized")
             return checks_results
 
+        if graph is None and self.graph_manager is not None and isinstance(self.graph_manager, GraphManager):
```

Comment:
the check here for `self.graph_manager` is not None can be removed as `isinstance` will be sufficient to catch it as well